azurelinuxagent/ga/env.py

```diff
@@ -99,85 +100,42 @@ def _get_dhcp_client_pid(self):
-            self._osutil.remove_legacy_firewall_rule(dst_ip=self._protocol.get_endpoint())
-            self._try_remove_legacy_firewall_rule = True
-
-        success, missing_firewall_rules = self._osutil.enable_firewall(dst_ip=self._protocol.get_endpoint(), uid=os.getuid())
-
-        if len(missing_firewall_rules) > 0:
-            if self._is_first_setup:
-                msg = "Created firewall rules for the Azure Fabric:\n{0}".format(self._get_firewall_state())
-                logger.info(msg)
-                add_event(op=WALAEventOperation.Firewall, message=msg)
-            else:
-                self._reset_count += 1
-                # We report immediately (when period is None) the first 5 instances, then we switch the period to every few hours
-                if self._report_period is None:
-                    msg = "Some firewall rules were missing: {0}. Re-created all the rules:\n{1}".format(missing_firewall_rules, self._get_firewall_state())
-                    if self._reset_count >= 5:
-                        self._report_period = datetime.timedelta(hours=3)
-                        self._reset_count = 0
-                        self._report_after = datetime.datetime.now() + self._report_period
-                elif datetime.datetime.now() >= self._report_after:
-                    msg = "Some firewall rules were missing: {0}. This has happened {1} time(s) since the last report. Re-created all the rules:\n{2}".format(
-                        missing_firewall_rules, self._reset_count, self._get_firewall_state())
-                    self._reset_count = 0
-                    self._report_after = datetime.datetime.now() + self._report_period
-                else:
-                    msg = ""
-                if msg != "":
-                    logger.info(msg)
-                    add_event(op=WALAEventOperation.ResetFirewall, message=msg)
-
-        add_periodic(
-            logger.EVERY_HOUR,
-            AGENT_NAME,
-            version=CURRENT_VERSION,
-            op=WALAEventOperation.Firewall,
-            is_success=success,
-            log_event=False)
-
-        self._is_first_setup = False
-
-    def _get_firewall_state(self):
         try:
-            return self._osutil.get_firewall_list()
+            if self._firewall_manager is None:
+                self._firewall_manager = FirewallManager.create(self._wire_server_address)
+
+            try:
+                if self._firewall_manager.check():
+                    return  # The firewall is configured correctly
+                self._report(event.warn, "The firewall has not been setup. Will set it up.")
+            except FirewallStateError as e:
+                self._report(event.warn, "The firewall is not configured correctly. {0}. Will reset it. Current state:\n{1}", ustr(e), self._firewall_manager.get_state())
+                self._firewall_manager.remove()
+            self._firewall_manager.setup()
```

Comment:
We no longer remove legacy rules here?

Reply:
removing the legacy rules is done only once, at initialization